src/tracer.js

```diff
@@ -254,23 +259,20 @@ export default class Tracer {
     }
 
     /**
-     *  Makes a concrete sampling decision for the ctx span context based on information
-     *  available to it's child span.
-     *
-     *  The ctx context's deferred mask is unset after this decision is made
+     * Reapplies sampling if the parent span isn't sampled
      *
+     * @param parentCtx the parent span context
      * @param ctx the span context
      * @param operationName the operation name for the child
      * @param tags tags to be applied by the sampler
      */
-    processDeferredSampling(ctx: SpanContext, operationName: string, tags: any) {
-        if (ctx.isDeferredSampling) {
+    upsampleChild(parentCtx: SpanContext, ctx: SpanContext, operationName: string, tags: any) {
+        if (!parentCtx.isSampled() && !parentCtx.isDebug()) {
             if (this._sampler.isSampled(operationName, tags)) {
                 ctx._flags |= constants.SAMPLED_MASK;
             } else {
                 ctx._flags &= ~constants.SAMPLED_MASK;
```

Comment:
I propose moving sampling from when a span is created to when the span is injected/finished/used as a parent and remove the notion of finalizing a sampling decision.

This change has the following advantages
- Simplifies the code by removing the redundant sampling logic in startSpan
- Upsampling logic isn't separated from the standard sampler. 
- Fixes metrics about spans sampled being incorrect because they are emitted based on the initial sampling decision
- Fixes the bug where a sampling decision is finalized the first time setOperation is called. (We can fix this independently, but I feel it's due to repeated code) 
- Simplifies integration when service names are tags in spans.

Apart from the refactoring required, I don't may big drawbacks. What are your thoughts on this?